core/src/mender-artifact.c

```diff
@@ -538,25 +639,25 @@ mender_artifact_read_manifest(mender_artifact_ctx_t *ctx) {
             mender_log_error("Invalid manifest file");
             return MENDER_FAIL;
         }
+        *separator = '\0';
 
-        /* Add checksum to the list */
-        mender_key_value_list_t *checksum = (mender_key_value_list_t *)calloc(1, sizeof(mender_key_value_list_t));
-        if (NULL == checksum) {
-            mender_log_error("Unable to allocate memory");
+        const char *checksum_str = line;
```

Comment:
Let's check that `checksum_str` is of the expected length. If not, there's no point in comparing it (and risking overflow).